src/dxtbx/model/detector.py

```diff
@@ -438,21 +445,32 @@ def from_phil(params, reference=None, beam=None):
             assert beam is not None
             set_mosflm_beam_centre(detector, beam, params.detector.mosflm_beam_centre)
 
-        # If the slow fast beam centre is set then update
-        if params.detector.slow_fast_beam_centre is not None:
+        fast_slow_beam_centre = None
+        if params.detector.slow_fast_beam_centre:
+            fast_slow_beam_centre = list(params.detector.slow_fast_beam_centre)
+            fast_slow_beam_centre[0], fast_slow_beam_centre[1] = (
+                fast_slow_beam_centre[1],
+                fast_slow_beam_centre[0],
+            )
+
+        if params.detector.fast_slow_beam_centre:
+            fast_slow_beam_centre = params.detector.fast_slow_beam_centre
+
+        # If the fast slow beam centre is set then update
+        if fast_slow_beam_centre is not None:
             panel_id = 0
-            if len(params.detector.slow_fast_beam_centre) > 2:
-                panel_id = params.detector.slow_fast_beam_centre[2]
+            if len(fast_slow_beam_centre) > 2:
```

Comment:
Is len() in {2, 3} the only valid options? I'd prefer an error in cases where the user has specified too many parameters

Reply:
this is handled at the PHIL stage via `.type = ints(size_min=2, size_max=3)`